src/Features/VisualBasic/Portable/ChangeSignature/VisualBasicChangeSignatureService.vb

```diff
@@ -391,47 +421,130 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.ChangeSignature
+
+            Dim newArguments = New List(Of ArgumentSyntax)()
+            Dim newSeparators = New SyntaxToken(arguments.Count - 1) {}
+            For newIndex = 0 To permutedArguments.Count - 1
+                Dim argument = permutedArguments(newIndex)
+                Dim originalIndex = argument.Index
 
-            Dim newArguments As List(Of IUnifiedArgumentSyntax) = MyBase.PermuteArguments(document, declarationSymbol, arguments.Select(Function(a) UnifiedArgumentSyntax.Create(a)).ToList(), permutedSignature, isReducedExtensionMethod)
+                Dim newParamNode = TransferLeadingTrivia(CType(DirectCast(argument, UnifiedArgumentSyntax), ArgumentSyntax), arguments(newIndex))
+                newArguments.Add(newParamNode)
 
-            Dim numSeparatorsToSkip = arguments.Count - newArguments.Count
-            Return SyntaxFactory.SeparatedList(newArguments.Select(Function(a) CType(DirectCast(a, UnifiedArgumentSyntax), ArgumentSyntax)), GetSeparators(arguments, numSeparatorsToSkip))
+                closeParenToken = TransferSeparatorTrivia(arguments, closeParenToken, permutedArguments.Count, newSeparators, newIndex, originalIndex)
+            Next
+
+            Return (SyntaxFactory.SeparatedList(newArguments, newSeparators.ToList().GetRange(0, If(permutedArguments.Count = 0, 0, permutedArguments.Count - 1))), closeParenToken)
         End Function
 
-        Private Function PermuteDeclaration(Of T As SyntaxNode)(list As SeparatedSyntaxList(Of T), updatedSignature As SignatureChange) As SeparatedSyntaxList(Of T)
+        Private Function PermuteDeclaration(Of T As SyntaxNode)(list As SeparatedSyntaxList(Of T), closeParenToken As SyntaxToken, updatedSignature As SignatureChange) As (permutedList As SeparatedSyntaxList(Of T), closeParenToken As SyntaxToken)
             Dim originalParameters = updatedSignature.OriginalConfiguration.ToListOfParameters()
             Dim reorderedParameters = updatedSignature.UpdatedConfiguration.ToListOfParameters()
 
-            Dim newParameters = New List(Of T)
-            For Each newParam In reorderedParameters
-                Dim pos = originalParameters.IndexOf(newParam)
-                Dim param = list(pos)
-                newParameters.Add(param)
+            Dim newParameters = New List(Of T)()
+            Dim newSeparators = New SyntaxToken(list.Count - 1) {}
+            For newIndex = 0 To reorderedParameters.Count - 1
+                Dim newParamSymbol = reorderedParameters(newIndex)
+                Dim originalIndex = originalParameters.IndexOf(newParamSymbol)
+
+                ' Copy whitespace trivia from original position.
+                Dim newParamNode = TransferLeadingTrivia(list(originalIndex), list(newIndex))
+                newParameters.Add(newParamNode)
+
+                closeParenToken = TransferSeparatorTrivia(list, closeParenToken, reorderedParameters.Count, newSeparators, newIndex, originalIndex)
```

Comment:
this looks like a lot of duplicated logic with C#.  we should try to unify.  We could potentially do that over two PRs.  one that unifies the logic between C# and VB for how things work today.  And then an update you make to that later to hae your new comment handling logic.

Reply:
The trivia logic between C# and VB seems quite different (imo) because of the lack of multi-line comments in VB, so I feel like unification may be difficult.
Re: unifying the logic between C# and VB for how they are today, could you possibly give an example of how the C# and VB logic could be unified? A lot of the duplication seems to be because of the different syntax constructs between C# and VB, so duplication seems fairly unavoidable...

Reply:
so i would have expected that we would basically design the algorithm so that it works on lists of nodes, tokens and comment trivia (which can be single-line or multi-line).  Then, when that algorithm runs on the VB code, it works the same, it just never runs into the 'multi-line comment' case.  

is there no common algorithm that would work with that concept? (there really may not be, i'm just not convinced currently).

Reply:
I agree a common algorithm would be best. However, the multi-line comment scenario changes the code significantly, so I believe that the amount of common logic is minimal. e.g.:
1) In C#, we modify the open paren token since it may have trailing trivia that should be associated with the first node (e.g., ```M(/*a */ a, b)```). This doesn't apply to VB.
2) In VB, we modify the close paren token to have the trailing trivia of the original separator, while in C# the trailing trivia is instead appended to the node, e.g. ```M(a, b, c /*comment */)``` vs. ```M(a, b, c) `comment```.
3) Due to the lack of multi-line comments in VB, we need to occasionally insert line breaks, e.g. when switching parameters a & b below: 
``` VB
M(a As Integer, ' a
b As Integer, c As Integer)
```

So I definitely see what you're saying, but at least in the context of this bug fix, I don't think that the amount of shared code (which is maybe a couple of lines spread throughout the fix) is enough to unify the logic.